crates/ruff_python_formatter/src/comments/mod.rs

```diff
@@ -138,6 +145,59 @@ impl SourceComment {
         DebugComment::new(self, source_code)
     }
 }
+
+/// The position of a comment in the source text.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub(crate) enum CommentTextPosition {
+    /// A comment that is on the same line as the preceding token and is separated by at least one line break from the following token.
+    ///
+    /// # Examples
+    ///
+    /// ## End of line
+    ///
+    /// ```python
+    /// a; # comment
+    /// b;
+    /// ```
+    ///
+    /// `# comment` is an end of line comments because it is separated by at least one line break from the following token `b`.
+    ///
+    /// ## Own line
```

Comment:
I got confused reading this -- I think we should remove this example from this specific docstring since it's already documented above `OwnLine`?